src/huggingface_hub/repocard_data.py

```diff
@@ -0,0 +1,369 @@
+                List of `huggingface_hub.EvalResult` that define evaluation results of the model. If provided,
+                `model_name` kwarg must be provided. Defaults to `None`.
+            model_name (`str`, *optional*):
+                A name for this model. Required if you provide `eval_results`. It is used along with
+                `eval_results` to construct the `model-index` within the card's metadata. The name
+                you supply here is what will be used on PapersWithCode's leaderboards. Defaults to None.
+            kwargs (`dict`, *optional*):
+                Additional metadata that will be added to the model card. Defaults to None.
+
+        Example:
+            >>> from huggingface_hub import CardData
+            >>> card_data = CardData(
+            ...     language="en",
+            ...     license="mit",
+            ...     library_name="timm",
+            ...     tags=['image-classification', 'resnet'],
+            ... )
+            >>> card_data.to_dict()
+            {'language': 'en', 'license': 'mit', 'library_name': 'timm', 'tags': ['image-classification', 'resnet']}
+
+        """
+
+        self.language = language
+        self.license = license
+        self.library_name = library_name
+        self.tags = tags
+        self.datasets = datasets
+        self.metrics = metrics
+        self.eval_results = eval_results
+        self.model_name = model_name
+        self.__dict__.update(kwargs)
+
+        if self.eval_results:
+            if type(self.eval_results) == EvalResult:
+                self.eval_results = [self.eval_results]
+            if self.model_name is None:
+                raise ValueError("`eval_results` requires `model_name` to be set.")
+
+    def to_dict(self):
+        """Converts CardData to a dict. It also formats the internal eval_results to
+        be compatible with the model-index format.
+
+        Returns:
+            `dict`: CardData represented as a dictionary ready to be dumped to a YAML
+            block for inclusion in a README.md file.
+        """
+
+        data_dict = copy.deepcopy(self.__dict__)
+        if self.eval_results is not None:
+            data_dict["model-index"] = eval_results_to_model_index(
+                self.model_name, self.eval_results
+            )
+            del data_dict["eval_results"], data_dict["model_name"]
+
+        return _remove_none(data_dict)
+
+    def to_yaml(self):
+        """Dumps CardData to a YAML block for inclusion in a README.md file."""
+        return yaml.dump(self.to_dict(), sort_keys=False).strip()
+
+    def __repr__(self):
+        return self.to_yaml()
+
+
+def model_index_to_eval_results(model_index: List[Dict[str, Any]]):
+    """Takes in a model index and returns a list of `huggingface_hub.EvalResult` objects.
+
+    A detailed spec of the model index can be found here:
+    https://github.com/huggingface/hub-docs/blob/main/modelcard.md
```

Comment:
Maybe https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1 is a better link